pybind and docopt added, ska fasta contig fix

johannahelene opened the pull request:
stable version of ska fasta, ska fasta was changed to account for contigs in fasta files, docent was added, pybind

johannahelene committed:
pybind and docopt added, ska fasta contig fix

stable version of ska fasta, ska fasta was changed to account for contigs in fasta files.

johnlees commented on ska/__main__.py:
```diff
@@ -1,55 +1,96 @@
-  -o <output>    Output prefix.
-  -l <file-list> File with a list of input files.
 """
+# -d --directory    Directory of fasta files (files have to end in *.fa or *.fasta)
```
If you want to leave this in, leave a comment explaining why (e.g. is it the eventual CLI you want to aim for?)

johnlees commented on ska/__main__.py:
```diff
@@ -1,55 +1,96 @@
-  ska map ref -l <file-list> -o <output>
-  ska (-h | --help)
-  ska (--version)
+  __main__.py --fasta <fasta>
+  __main__.py --align <align>
+  __main__.py --map <map>
```
```suggestion
  ska --fasta <fasta>
  ska --align <align>
  ska --map <map>
```

johannahelene committed:
ska_fasta and ska_align two files now

ska_fasta and ska_align were taken apart and are in seperate files now

johannahelene committed:
Smaller Update

Added openMP, chrono, changed the input files to be in style of rfiles from PopPUNK

johannahelene committed:
added unittest for ska_align

ska_align unittest with small sequence to test

johannahelene committed:
Create GitHub action for testing

add action for testing ska_align.py

johannahelene committed:
Update python-package-conda.yml

johannahelene committed:
Update python-package-conda.yml

johannahelene committed:
Minor edits

johannahelene committed:
Merge branch 'test_action' into johanna_dev

johannahelene committed:
Merge branch 'test_action' into johanna_dev

johannahelene committed:
Merge remote-tracking branch 'origin/johanna_dev' into johanna_dev

johannahelene committed:
Add environment file

johannahelene committed:
Update python-package-conda.yml

johannahelene committed:
Update intel_ska.yml

johannahelene committed:
changes paths for testing

johannahelene committed:
added alignment file

johannahelene committed:
paths again

johannahelene committed:
Update variant_alignment.aln

johannahelene committed:
paths

johannahelene committed:
Merge remote-tracking branch 'origin/johanna_dev' into johanna_dev

johannahelene committed:
Update 10_test_cluster.txt

johannahelene committed:
Update 10_test_cluster.txt

johannahelene committed:
paths

johannahelene committed:
Merge remote-tracking branch 'origin/johanna_dev' into johanna_dev

johannahelene committed:
paths

johannahelene committed:
Added functions

added progressbar and writing out to file as an ascii bitstream

johannahelene committed:
Added functionality to ska fasta

started on bit string, added ascii_bitstring, added k-mer length constraint

johannahelene committed:
bit vector for kmer string

added function to convert current kmer to a bitvector

johannahelene committed:
Added bitvectors, rolling k-mers

bitvectors for both the kmer and the value bases, three functions for k-mer creation
